apps/qubit/modules/contactinformation/actions/indexAction.class.php

```diff
@@ -21,15 +21,15 @@ class ContactInformationIndexAction extends sfAction
 {
     public function execute($request)
     {
+        $this->resource = QubitContactInformation::getById($request->id);
+
         $hasEditAccess = QubitAcl::check($this->resource, 'update')
             || $this->getUser()->hasGroup(QubitAclGroup::EDITOR_ID);
         // Check user authorization
         if (!($this->getUser()->isAuthenticated() && $hasEditAccess)) {
             QubitAcl::forwardUnauthorized();
         }
```

Comment:
If I understand correctly, `QubitAcl::check()` will now check if the user is authenticated AND if the user is an admin or an editor. So we could simplify this entire block to:

```php
if (!QubitAcl::check($this->resource, 'update')) {
    QubitAcl::forwardUnauthorized();
}
```

Also, `update` in an index action?

Reply:
ah, right. thanks, @jraddaoui !